sled-agent/src/sim/server.rs

```diff
@@ -100,7 +100,7 @@ impl Server {
                             .hardware
                             .hardware_threads,
                         usable_physical_ram: NexusTypes::ByteCount::try_from(
-                            config.hardware.physical_ram,
+                            config.hardware.physical_ram as i64,
```

Comment:
The `as i64` around ByteCount seems unfortunate.  (not saying it's a blocker.)  Part of the point of that type was to make it hard to accidentally confuse bytes for mebibytes or megabytes or whatever by having explicit constructors and getters.  If we're just throwing an i64 into a constructor, it seems like we're losing that (and making it easy to introduce that kind of problem).

I don't know what the answer is.  Could/should this be one of those types that we actually share across client/server, as a sort of primitive?